src/app/select-session/page.tsx

```diff
@@ -0,0 +1,228 @@
+      if (card.status === SessionCardStatus.SELECTED) {
+        return true;
+      } else {
+        return false;
+      }
+    }).length;
+    setSessionsSelected(numberActive);
+  }, [session]);
+
+  /**Doesn't show bouncing arrow on load if there is no overflow */
+  useEffect(() => {
+    setTimeout(() => {
+      if (ref.current && ref.current.scrollHeight > ref.current.clientHeight) {
+        setIsOverflown(true);
+      }
+    }, 200);
+  });
+
+  return (
+    <div className="h-[100dvh] flex flex-col">
+      <div className="pt-5 pb-5 pl-7 flex">
+        <Heading>Sessions</Heading>
+        <CgProfile
+          className="ml-auto mr-4"
+          size={30}
+          onClick={() => alert("profile")}
+        ></CgProfile>
+      </div>
+      <div
+        className={twJoin(
+          "bg-[#EAEEF3] min-h-[66px] flex",
+          !isMember && "justify-center",
+        )}
+      >
+        <p className="p-5 text-md flex items-center font-medium">
+          Hey {isMember ? firstName : "Guest"}!
+        </p>
+        {isMember && (
+          <p className="text-xs flex flex-row-reverse grow items-center">
+            Prepaid Sessions <br />
+            Remaining
+          </p>
+        )}
+        {isMember && (
+          <p className="flex m-4 justify-center items-center rounded bg-[#D9D9D9] h-[35px] w-[35px]">
+            {remainingSessions}
+          </p>
+        )}
+      </div>
+      <div
+        className={twJoin(
+          "flex min-h-[76px] p-5 items-center",
+          !isMember && "justify-center text-center",
+        )}
+      >
+        <p className={`max-w-[70%] text-s leading-5 font-medium`}>
+          Please select a badminton session for this week
+        </p>
+        {isMember && (
+          <div className="flex grow flex-row-reverse">
+            <p
+              className={twJoin(
+                "flex items-center justify-center rounded bg-[#D9D9D9] h-[34px] w-[63px] font-semibold",
+                shake &&
+                  "error-shake text-[#AF3737] border-[#AF3737] border-solid border",
+              )}
+            >
+              {sessionsSelected} / {maxSessions}
+            </p>
+          </div>
+        )}
+      </div>
+
+      {/* TODO: check whether to use this or mask */}
+      <ScrollShadow>
+      <div
+        // scroll-fade py-4
+        className="flex flex-col overflow-y-auto w-full h-[calc(100dvh-329px)] gap-3 px-5"
+        onScroll={(e) => setScrollIndicator(false)}
+        ref={ref}
+      >
+        {session.map((card, index) => {
+          return (
+            <SessionCard
+              startDate={card.startDate}
+              endDate={card.endDate}
+              location={card.location}
+              status={card.status}
+              key={index}
+              onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                sessionClick(e, index)
+              }
+            ></SessionCard>
+          );
+        })}
```

Comment:
```suggestion
        {session.map((card) => {
          return (
            <SessionCard
              startDate={card.startDate}
              endDate={card.endDate}
              location={card.location}
              status={card.status}
              key={card.id}
              onChange={(e: ChangeEvent<HTMLInputElement>) =>
                sessionClick(e, index)
              }
            ></SessionCard>
          );
        })}
```

Please use id, Whether they are mocked (faked) or real id from the database.

Reply:
React is going to be doing wild processing and slowing the app down if you use index as key